tests/cpp/outer_loop/outer_loop_test.cpp

```diff
@@ -0,0 +1,272 @@
+
+// #include "MasterUpdate.h"
+
+#include "LoggerFactories.h"
+#include "MasterUpdate.h"
+#include "OuterLoopInputDataReader.h"
+#include "WriterFactories.h"
+#include "gtest/gtest.h"
+#include "multisolver_interface/environment.h"
+
+int my_argc;
+char** my_argv;
+
+boost::mpi::environment* penv = nullptr;
+boost::mpi::communicator* pworld = nullptr;
+
+using namespace Outerloop;
+
+int main(int argc, char** argv) {
+  ::testing::InitGoogleTest(&argc, argv);
+
+  mpi::environment env(my_argc, my_argv);
+  mpi::communicator world;
+  penv = &env;
+  pworld = &world;
+  my_argc = argc;
+  my_argv = argv;
+  return RUN_ALL_TESTS();
+}
+
+//-------------------- MasterUpdateBaseTest -------------------------
+const auto STUDY_PATH =
+    std::filesystem::path("data_test") / "external_loop_test";
+const auto LP_DIR = STUDY_PATH / "lp";
+const auto OPTIONS_FILE = LP_DIR / "options.json";
+const auto OUTER_OPTIONS_FILE = LP_DIR / "outer_loop_options.yml";
+
+class MasterUpdateBaseTest : public ::testing::TestWithParam<std::string> {
+ public:
+  pBendersBase benders;
+  std::shared_ptr<MathLoggerDriver> math_log_driver;
+  Logger logger;
+  Writer writer;
+
+  void SetUp() override {
+    math_log_driver = MathLoggerFactory::get_void_logger();
+    logger = build_void_logger();
+    writer = build_void_writer();
+  }
+  BendersBaseOptions BuildBendersOptions() {
+    SimulationOptions options(OPTIONS_FILE);
+    return options.get_benders_options();
+  }
+};
+
+auto solvers() {
+  std::vector<std::string> solvers_name;
+  solvers_name.push_back("COIN");
+  if (LoadXpress::XpressLoader xpressLoader;
+      xpressLoader.XpressIsCorrectlyInstalled()) {
+    solvers_name.push_back("XPRESS");
+  }
+  return solvers_name;
+}
+
+INSTANTIATE_TEST_SUITE_P(availsolvers, MasterUpdateBaseTest,
+                         ::testing::ValuesIn(solvers()));
+double LambdaMax(pBendersBase benders) {
+  const auto& obj = benders->MasterObjectiveFunctionCoeffs();
+  const auto max_invest = benders->BestIterationWorkerMaster().get_max_invest();
+  double lambda_max = 0;
+  for (const auto& [var_name, var_id] : benders->MasterVariables()) {
+    lambda_max += obj[var_id] * max_invest.at(var_name);
+  }
+  return lambda_max;
+}
+
+void CheckMinInvestmentConstraint(const VariableMap& master_variables,
+                                  const std::vector<double>& expected_coeffs,
+                                  const double expected_rhs, char expected_sign,
+                                  const std::vector<double>& coeffs,
+                                  const double rhs, char sign) {
+  for (auto const& [name, var_id] : master_variables) {
+    ASSERT_EQ(expected_coeffs[var_id], coeffs[var_id]);
+  }
+
+  ASSERT_EQ(expected_rhs, rhs);
+  ASSERT_EQ(expected_sign, sign);
+}
+
+TEST_P(MasterUpdateBaseTest, ConstraintIsAddedBendersMPI) {
+  BendersBaseOptions benders_options = BuildBendersOptions();
```

Comment:
Is it possible to run Benders from `LP_DIR` ? And even in a temp directory to avoid modifying files in the repo when launching the test ?

Reply:
It’s possible we have to modify the test working dir to lp, doing so shadows the mechanism of how Benders retrieves his input/output files etc…
I prefer The 2nd sol ;)

Reply:
Ok